### PR TITLE
Adding client usage extension for server cert (#305)

### PR DESCRIPTION
**Pull request by mmerrill3**
PR to address issue (#305 )
Signed-off-by: mmerrill3 <michael.merrill@vonage.com>

### CODE REVIEW

**Comment by mmerrill3** [comment]
I had issues with make test.  Don't think its related to this change

**Comment by mmerrill3** [comment]
@justinsb I made the grpc sevices' cert (etcd-manager's service), and the etcd server's cert both available for using as a client cert during mutual TLS.  Not sure if you have objections to allowing the grpc service's certificate being used in this way.

File: pkg/etcd/pki.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (p *etcdProcess) createKeypairs(peersCA *pki.Keypair, clientsCA *pki.Keypai
 
 		certConfig := certutil.Config{
 			CommonName: me.Name,
-			Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+			Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
```
Comment (justinsb) [inline pkg/etcd/pki.go]: Nit: A comment with a link to e.g. https://github.com/etcd-io/etcd/issues/9785 would be helpful for people scratching their head about this one in future :-)

File: pkg/tlsconfig/options.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func GRPCServerConfig(keypairs *pki.Keypairs, myPeerID string) (*tls.Config, err
 
 	config := certutil.Config{
 		CommonName: "etcd-manager-server-" + myPeerID,
-		Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
```
Comment (justinsb) [inline pkg/tlsconfig/options.go]: Do we need client here?  I think we set up our client certs separately (above in GRPCClientConfig)

**Comment by justinsb** [comment]
Thanks for figuring this out @mmerrill3 .  I don't particularly object to adding the client flag, except that I don't think we need it - was there a reason to add it?

**Comment by mmerrill3** [comment]
@justinsb I put it in there to figure out which of the server certs was being used as a client cert, since I wasn't sure at first about which server cert was used where in etcd-manager.  The gRPC service one for etcd-manager was changed first in my testing, and I just didn't undo it.  If there's no need for the client attribute on it, I'll remove it.  I wasn't sure, which is why I was asking if there was any use case where this cert was used as a client cert.

I'll update my PR tonight.

**Comment by justinsb** [comment]
That's great @mmerrill3 - thanks again!  Please ping the PR when you push the change so I can (be more likely to) notice it :-)

File: pkg/etcd/pki.go (hunk 1 above)
Comment (mmerrill3) [reply pkg/etcd/pki.go]: added a comment with a link to the original issue

File: pkg/tlsconfig/options.go (hunk 2 above)
Comment (mmerrill3) [reply pkg/tlsconfig/options.go]: I just removed it, its not really needed per comments.

**Comment by mmerrill3** [comment]
@justinsb the PR has been updated and commits squashed down.

**Comment by justinsb** [comment]
Awesome @mmerrill3 - thank you so much!

/approve
/lgtm

**Pull request merged**